### ISSUE TITLE
What kind of 'discrete' variables are initialized to pre-values in algorithms?

### ISSUE THREAD

**Issue by henrikt-ma** [modelica/ModelicaSpecification#2640]
In [11.1.2 _Execution of an algorithm in a model_](https://specification.modelica.org/master/statements-and-algorithm-sections.html#execution-of-an-algorithm-in-a-model) we have:
> A discrete variable `v` is initialized with `pre(v)`.

What kind of _discrete_ (variable) is meant here?
- A discrete-time variable.
- A variable assigned in a when-clause.  (Subset of all discrete-time variables.)
- A variable declared `discrete`.  (Subset of all variables assigned in a when-clause.  This alternative appears the most unlikely to me.)

For illustration, consider this model:
```
model StartOrPreAlgorithmInit
  Integer x(start = 1, fixed = false) "Discrete-time variable not assigned in when-clause";
algorithm
  if 1 < time and time < 2 then
    x := 2;
  end if;
  /* What will x be after time 2? */
end StartOrPreAlgorithmInit;
```

**Comment by henrikt-ma** [comment]
To me, the alternative which makes most sense is the first one; if `pre` can be applied, do it – otherwise, use `start`.

**Comment by henrikt-ma** [comment]
Maybe part of the answer lies in how the we handle `fixed = true` in section 8.6?  It also just says _discrete_, but in a context where it is clear that the alternative is continuous-time.

**Comment by HansOlsson** [comment]
> Maybe part of the answer lies in how the we handle `fixed = true` in section 8.6? It also just says _discrete_, but in a context where it is clear that the alternative is continuous-time.

Yes. I believe that is the answer, and it seems we have a number of uses of "discrete variable" that should be "discrete-time variable".
